### PR TITLE
Allow sparse rhs to be input as SparseVector

### PR DESCRIPTION
**Pull request by Pbellive**
Currently need to input an n X 1 `SparseMatrixCSC` for single sparse rhs. Also added precompilation.

### CODE REVIEW

File: src/MUMPSfuncs.jl [hunk 1]
```diff
@@ -116,6 +116,19 @@ function applyMUMPS!(factor::MUMPSfactorization{Float64},rhs::SparseMatrixCSC{Fl
 	return x
 end
 
+function applyMUMPS!(factor::MUMPSfactorization{Float64},rhs::SparseVector{Float64}, x::Array{Float64,1},tr::Int=0)
+	nrhs = 1
+	nzrhs = nnz(rhs)
+	colptr = [1;3]
```
Comment (Sacha0) [inline src/MUMPSfuncs.jl]: Should this `colptr` match the `colptr` for a `SparseMatrixCSC` representation of `rhs::SparseVector`? If so, shouldn't `colptr = [1, nzrhs+1]`? If not, kindly ignore :).

**Comment by Pbellive** [comment]
You're right of course @Sacha0. The lines you flagged should have been
```
colptr = [1; nzrhs + 1]
```
I pushed the wrong thing. Sorry about that and thanks for pointing it out. I fixed the error, added tests, and organized the tests in to sets. Think this should be good to go now.

**Comment by Pbellive** [comment]
Is everyone ok to merge this now? I'm happy to make changes if needed but if people are happy with it as is, it would be great to get this merged. @lruthotto are you currently the only person with permission to merge PRs for this repo?

Thanks!

**Comment by lruthotto** [comment]
Sorry for the wait, Patrick. This looks fabulous (as usual). I think everyone in JuliaSparse can merge this.

**Pull request merged**